docs/user_guide/docs_guidelines.rst

```diff
@@ -11,6 +11,74 @@ To contribute to it, open a pull request with the changes to
 :external+doctools:ref:`docs_guidelines` first **and** the additional guidelines
 below specific to the HDL repository.
 
+Before creating a new page
+--------------------------------------------------------------------------------
+
+This section has the sole role to present the steps that are required to be
+able to create and build a new documentation related page. Some steps are
+recommended to be revised regularly to keep the necessary tools updated. 
+
+First, make sure you have the latest version of ``pip`` installed. It must be
+newer than 23 version. If not, update it by running the following command:
+
+.. code-block::
+
+   pip install pip --upgrade
+
+Than install the necessary documentation tools by running (assuming the :git-hdl:`HDL <>`
+repository is the working directory):
+
+.. code-block::
+   
+   cd docs/
+   pip install -r requirements.txt --upgrade
+
+It's recommended to regularly update the documentation tools. This is done by
+running the following command:
+
+.. code-block::
+   
+   cd docs/
+   pip install -r requirements.txt --upgrade
+
+Before building a page, it's recommended to build all the projects from
+``/library``. Some references (files used in specific parts of the page) are
+directly taken from the libraries' project folder (e.g.: ``/library/axi_dmac/component.xml``)
+after being built. Build the libraries by running:
+
+.. code-block::
+   
+   cd library/
+   make
+
+Now, after the page has been written, inside ``/docs`` folder run the following
+command:
+
+.. code-block::
+   
+   cd docs/
+   make html
+
+The generated documentation will be available at ``/docs/_build/html``.
+
+It's recommended to clean the cached data when rebuilding a page and major
+changes have been made. This is done by running the below commands (inside
+``/docs`` folder):
```

Comment:
```suggestion
It's recommended to clean the cached data when changing the document structure,
like adding a new page, or other major changes have been made.
This is because Sphinx rebuilds only "touched" pages and, for example, 
adding a page changes the sidebar navigation for all pages.
This is done by running the below commands (inside ``/docs`` folder):
```